pod.go

```diff
@@ -391,6 +395,9 @@ func (p *Pod) Annotations(key string) (string, error) {
 
 // SetAnnotations sets or adds an annotations
 func (p *Pod) SetAnnotations(annotations map[string]string) error {
+	annotationsLock.Lock()
```

Comment:
We want this lock to be part of the pod handler. In case you have several pods, you don't want to wait the lock for another pod's annotations.